packages/rules/src/required.ts

```diff
@@ -6,8 +6,10 @@ import { FormKitValidationRule } from '@formkit/validation'
  * @param context - The FormKitValidationContext
  * @public
  */
-const required: FormKitValidationRule = function required({ value }) {
-  return !empty(value)
+const required: FormKitValidationRule = function required({ value }, action = 'default') {
+  return action === 'trim'
```

Comment:
One thought I had here — often the values are not strings, they may be arrays, or objects etc. Perhaps we should do a little check for that here instead of casting to a string:

```suggestion
  return action === 'trim' && typeof value === 'string' ? !empty(value.trim()) : !empty(value)
```

Reply:
I thought I added that before I pushed. I even said something to myself when updating the docs because it's not a text field example. I've updated accordingly.